### DISCUSSION TITLE
SANDBOX

### DISCUSSION THREAD

**Discussion by swbittib** [fattureincloud/api#156] [Q&A] [answered]
Buongiorno,

in che modo si possono testare le API prima di andare in produzione?

Grazie

**Comment by valmoz** [comment, marked as answer]
Buongiorno,

Per quanto riguarda ambiente di playground/sandbox/di test, siamo già a conoscenza della problematica. Purtroppo non abbiamo una tempistica di risoluzione al momento.
Per come sono state definite le API è però possibile procedere con la creazione/cancellazione delle risorse senza problemi, permettendo di testare i diversi metodi API senza compromettere l'account. Molti nostri utenti hanno comunque deciso di creare un secondo account FIC di TEST che utilizzano per testare il codice senza sporcare l'account principale. Per evitare di dover acquistare una seconda licenza, è possibile richiedere al supporto clienti una licenza di prova per fini di developent e testing.
L'unico metodo che può effettivamente creare problemi è il [Send E-Invoice](https://developers.fattureincloud.it/docs/guides/e-invoice-management#-send-the-e-invoice), cioé il metodo utilizzato per inviare materialmente i documenti al SDI: in tal caso è necessario utilizzare il metodo con cautela, considerando però che:

- Se la fatturazione elettronica è disabilitata sull'account, il documento non viene comunque inviato all'SDI
- Se l'account ha una licenza di prova, non invia la fattura
- Stiamo comunque studiando una soluzione per aggirare il problema, ad esempio permettendo di utilizzare il metodo in modalità Dry Run

Resto a disposizione per ulteriori domande in merito.
Mauro

**Reply by valmoz** [reply]
Ciao @swbittib 
abbiamo rilasciato il [Dry Run flag](https://developers.fattureincloud.it/blog/dry-run)

Buona giornata
Mauro
